### ISSUE TITLE
Build failure: KMonad

### ISSUE THREAD

**Issue by jfvillablanca** [NixOS/nixpkgs#224190]
### Steps To Reproduce

1. `nix build -L -v --show-trace .#haskellPackages.kmonad`

### Build log

```
error: Package ‘kmonad-0.4.1’ in /nix/store/g8ajs4xv5yrf6pz033xbif6ilwwjl85a-source/pkgs/development/haskell-modules/hackage-packages.nix:172214 is marked as broken, refusing to evaluate.
```
Link to [full log](https://gist.github.com/jfvillablanca/ce7625cf75316d39b5a0a9a4e3114937)

### Additional context

This issue is discussed in #84550

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/x6yjkzgmdppwi95y4gvm4lxi6q27936s-nix-info
copying path '/nix/store/x6yjkzgmdppwi95y4gvm4lxi6q27936s-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.20, NixOS, 23.05 (Stoat), 23.05.20230321.19cf008`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by sternenseemann** [comment]
Seems like kmonad needs a new release to stackage. We use Stackage LTS 20 which upstream [supports](https://github.com/kmonad/kmonad/commit/b9798a78ae2c060a03695d80168edc1c3a81a066) on its development branch, but the currently released version on hackage has outdated bounds:

```
base >=4.12.0.0 && <4.13,
lens >=4.17.1 && <4.18,
megaparsec >=7.0.5 && <7.1,
optparse-applicative >=0.14.3.0 && <0.15,
time >=1.8.0.2 && <1.9
```

@slotThe Would it be possible to upload a new release to hackage?

**Comment by slotThe** [comment]
@sternenseemann unfortunately, I don't have access to kmonad on Hackage (or I would have revbumped it a while ago). I will try to contact David and see whether it could be arranged to give me the necessary rights

**Comment by slotThe** [comment]
@jfvillablanca @sternenseemann I'm terribly sorry that this took as long as it did, but I have just released [0.4.2](https://hackage.haskell.org/package/kmonad-0.4.2) on Hackage, which should hopefully fix the problem.

**Comment by slotThe** [comment]
@jfvillablanca Done in https://github.com/NixOS/nixpkgs/pull/267450 :)
